Jenkinsfile

```diff
@@ -0,0 +1,68 @@
+def testPassed = true
+
+pipeline {
+  agent any
+  environment {
+    ARTIFACT = "artifact.zip"
+  }
+  stages {
+    stage('Build') {
+      steps {
+        sh "dotnet restore"
+        sh "dotnet build --configuration Release"
+        script {
+          try {
+            sh "dotnet test --logger \"trx;LogFileName=TestResults.trx\""
+          } catch (ex) {
+            unstable('Some tests failed')
+            testPassed = false
+          }
+        }
+        mstest testResultsFile:"**/*.trx", keepLongStdio: true
+        sh "cd Lavinia-api/bin/Release/netcoreapp3.1; zip -r ${WORKSPACE}/${ARTIFACT} *; cd ${WORKSPACE}"
+        archiveArtifacts artifacts: ARTIFACT
+      }
+    }
+
+    stage('Deploy') {
+      when {
+        allOf {
+          tag "*.*.*"
+          expression { testPassed }
+        }
+      }
+      steps {
+        ansiblePlaybook(
+          playbook: '/storage/api_deploy.yaml',
+          credentialsId: 'ansible_key',
+          inventory: '/storage/hosts',
+          disableHostKeyChecking: true
```

Comment:
What's this madness? This is not the kind of thing that stands up to a security review.

Reply:
When you use SSH and don't check the host key, you have in principle no idea what computer you just connected to.